### PR TITLE
refactor: create function to set everything up before calling docfx

### PR DESCRIPTION
**Pull request by tbpg**
It would be nice to be able to test the docfx.json generation directly.
So, I split that into a separate function and added a test for it.

A follow up of this will be automatically adding xref files to the
docfx.json. This change will allow that to be sufficiently tested.

### CODE REVIEW

**Review by dandhlee — APPROVED** [review]
Looks good!

File: tests/test_generate.py [hunk 1]
```diff
@@ -138,21 +144,44 @@ def verify_content(html_blob, tmpdir):
...
+    bucket, yaml_blob, html_blob = setup_testdata(
+        yaml_dir, storage_client, credentials, test_bucket
+    )
+
+    tmp_path = pathlib.Path(tempfile.TemporaryDirectory(prefix="doc-pipeline.").name)
+    metadata_path = generate.setup_docfx(tmp_path, yaml_blob)
+
+    docfx_json_file = tmp_path.joinpath("docfx.json")
+    assert docfx_json_file.exists()
+    with open(docfx_json_file) as w:
```
Comment (jskeet) [inline tests/test_generate.py:169]: We could presumably have a golden docfx.json as well, instead of just checking one bit?

**Review by jskeet — APPROVED** [review]
Can't say I've followed every little bit of it, but generally happy with the thrust.

File: tests/test_generate.py (hunk 1 above)
Comment (tbpg) [reply tests/test_generate.py:169]: I played around with this a bit, but didn't find a satisfactory way to test it that actually felt like it was testing more than the other `docfx.json` test we currently have. So, going to leave as-is unless you object.

**Review by tbpg — COMMENTED** [review]
Thank you!

**Comment by tbpg** [comment]
Moving forward with merging so I can get xrefs working end to end.

**Pull request merged**